lib/types/src/compilation/symbols.rs

```diff
@@ -176,7 +172,10 @@ impl SymbolRegistry for ModuleMetadataSymbolRegistry {
     }
 
     fn name_to_symbol(&self, name: &str) -> Option<Symbol> {
-        if let Some(index) = name.strip_prefix(&format!("wasmer_function_{}_", self.prefix)) {
+        if let Some(prefix) = name.strip_prefix("WASMER_METADATA_") {
```

Comment:
```suggestion
        if name == "WASMER_METADATA_{self.prefix}" {
```

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3430/commits/b963d2cc54d6048b87ba328e66f1c544b96f6e28